src/pdfz.cpp

```diff
@@ -323,6 +369,70 @@ void apply_systematic(const SystematicDescriptor* syst,
   }
 }
 
+///// EvalHist kernels
+HEMI_DEV_CALLABLE_INLINE
+void apply_oscillation_systematic(const SystematicDescriptor* syst,
+				  double* fields, const double* parameters,
+				  const int param_stride,
+				  const double* pee_osc_lut, const int pee_osc_lut_size,
+				  const double* pee_osc_lut_pars, const int pee_osc_lut_npars,
+				  int &reject_event, RNGState* rng) {
+  if (pee_osc_lut_size == 0) {
+    printf("pdfz::apply_systematic: Something is seriously wrong with the look up table!\n");
+    return;
```

Comment:
Since there are some really bad conditions here that `return`, I wonder if it'd be worth setting some error flag that gets checked when samples are periodically copied back to the host memory, and `assert`ing at that point.